client/controller/controller.js

```diff
@@ -0,0 +1,355 @@
+      document.getElementById('slideshow').style.visibility = "visible";
+
+
+      var service = new google.maps.places.PlacesService(map.gMap);
+      var listOfRestaurants = $scope.filterdata;
+
+
+      var id = listOfRestaurants[0].place_id;
+
+      console.log(listOfRestaurants[0]);
+
+        service.getDetails({
+          placeId: id
+        }, function(result, status){
+          if(status === google.maps.places.PlacesServiceStatus.OK){
+            console.log('results below');
+            console.log(result);
+
+
+            $scope.restaurantName = result.name;
+
+            $scope.callServer();
+
+            //display website
+            $scope.website        =  result.website;
+
+
+            if(result.reviews){
+              console.log('reviews are ' + result.reviews.length);
+               //document.getElementById('accordion').style.visibility = "visible";
+              //store reviews in scope variable
+              $scope.reviews      = result.reviews;
+
+
+            }
+
+
+          if(result.photos){
+            $scope.listOfPhotos = [];
+            for(var i=0; i < result.photos.length; i++){
+              var photos = result.photos[i].getUrl({'maxWidth': 600, 'maxHeight': 600});
+
+             $scope.listOfPhotos.push(photos);
+             //$scope.test = $scope.photos
+
+            }
+          }
+          else{
+            document.getElementById('slideshow').style.visibility= "hidden";
+            $scope.noPhotoText = "No photos to display"
+          }
+
+
+
+
+          } //end of service if
+
+
+        })
+
+
+    } //end of getDetail function.
+
+
+
+      //<-----------This is the function for the next button-------->
+    $scope.next = function(){
+
+
+
+      $scope.listOfPhotos = [];
+
+      var restaurant = $scope.filterdata;
+
+      $scope.reset();
+
+
+      if(restaurant.length <= 1){
+        alert('no more restaurants. Search again');
+        $scope.reset();
+      }
+
+      else{
+
+          $scope.filterdata.splice(0, 1);
+
+
+
+          restaurant = $scope.filterdata;
+
+
+          $scope.getDetails();
+
+          var server = {location: $scope.place, restaurant: restaurant[0].name};
+
+          $http.post("/getyelp", server)
+            .success(function(response){
+
+              var restaurant = response.businesses;
+
+             //Storing all relevent data
+            var name            = restaurant[0].name
+
+
+            localStorage.setItem('restaurantName', name);
+
+            var rating          = restaurant[0].rating;
+            var image           = restaurant[0].image_url;
+            var url             = restaurant[0].url;
+            var status          = restaurant[0].is_closed;
+            var contact         = restaurant[0].phone;
+            $scope.rating_img   = restaurant[0].rating_img_url;
+            var address         = restaurant[0].location.display_address;
+
+              localStorage.setItem('restaurant', address);
```

Comment:
Put it at the end or call a function to setItem or removeItem from localStorage.